beesdoo_shift/wizard/subscribe.py

```diff
@@ -121,10 +121,16 @@ def subscribe(self):
         if self.reset_compensation_counter:
             data['sc'] = 0
 
-        status_id = self.env['cooperative.status'].search([('cooperator_id', '=', self.cooperator_id.id)])
+        coop_status_obj = self.env['cooperative.status']
+        status_id = coop_status_obj.search(
+            [('cooperator_id', '=', self.cooperator_id.id)])
         if status_id:
             status_id.sudo().write(data)
         else:
-            self.env['cooperative.status'].sudo().create(data)
-            
-
+            status_id = coop_status_obj.sudo().create(data)
+            # Normally the write method is not necessary here.
+            # But it does not work without it. You have to make 2 registration
```

Comment:
Maybe it would be better to investigate the root reason why this does not work instead of writing a piece of code that doesn't seems to make sense

Reply:
We did investigate but it took a lot of time. We did prefer to do it like this to not exceed the estimate time. I can maybe review it again with Rémy when he comes back.

Reply:
Yes in fact, we needed to fix it and the dedicated budget time to investigate and fix was consumed. So we chosen to fix it like this and to put a todo to investigate it later during a benevolent shift.

If you feel to investigate this Thibaut, we'll be so greatful.